src/components/Toast/Toast.tsx

```diff
@@ -100,36 +139,83 @@ const Toast: FC<ToastProps> & { types?: typeof ToastType; actionTypes?: typeof T
     };
   }, [open, autoHideDuration, setAutoHideTimer]);
 
-  const iconElement = !hideIcon && getIcon(type, icon);
+  const iconElement =
+    iconSize === "small"
+      ? !hideIcon && getIcon(type, icon, ToastIconSize.SMALL)
+      : !hideIcon && getIcon(type, icon, ToastIconSize.LARGE);
 
   return (
     <CSSTransition in={open} classNames="l3-style-toast-animation" timeout={400} unmountOnExit>
-      <div className={classNames} role="alert" aria-live="polite">
-        {iconElement && <div className="l3-style-toast-icon">{iconElement}</div>}
-        <div
-          className={cx("l3-style-toast-content", {
-            "l3-style-toast-content-no-icon": !iconElement
-          })}
-        >
-          {children}
-          {toastLinks}
-        </div>
-        {(toastButtons || deprecatedAction) && (
-          <div className="l3-style-toast-action">{toastButtons || deprecatedAction}</div>
-        )}
-        {closeable && (
-          <Button
-            className="l3-style-toast_close-button"
-            onClick={handleClose}
-            size={Button.sizes.SMALL}
-            kind={Button.kinds.TERTIARY}
-            color={Button.colors.ON_PRIMARY_COLOR}
-            ariaLabel="close-toast"
+      <>
+        <div className={classNames} role="alert" aria-live="polite">
+          {avatar && (
+            <div className="l3-style-toast-avatar">
+              <Avatar
+                type={
+                  (avatarType === "img" && Avatar.types.IMG) ||
+                  (avatarType === "icon" && Avatar.types.ICON) ||
+                  (avatarType === "text" && Avatar.types.TEXT)
+                }
+                size={
+                  (avatarSize === "small" && Avatar.sizes.SMALL) ||
+                  (avatarSize === "large" && Avatar.sizes.LARGE) ||
+                  (avatarSize === "medium" && Avatar.sizes.MEDIUM)
+                }
+                src={avatarSrc}
+                icon={avatarIcon}
+                text={avatarText}
+                ariaLabel={avatarAriaLabel}
+                role={avatarRole}
+                rectangle={avatarRectangle}
+                square={avatarSquare}
+                isDisabled={avatarDisabled}
+                withoutBorder={avatarBorder}
+                textClassName={avatarTextClassName}
+                backgroundColor={avatarBackgroundColor}
+                className={avatarClassName}
+                id={avatarId}
+                data-testid={avatarDataTestId}
+              />
+            </div>
+          )}
+          {(!artWork || !avatar) && iconSize === "small" ? (
```

Comment:
and medium? please create it dynamically  not use static checks

Reply:
@DavitBakhutashvili Also need check Tests